frontend/src/router/index.js

```diff
@@ -135,4 +142,17 @@ const router = createRouter({
   },
 })
 
+router.beforeEach((to, from, next) => {
+  const hasAnyPermission = store.getters['auth/hasAnyPermission']
+  if (hasAnyPermission || isUnauthorizedComponent(to)) {
+    next()
+  } else {
+    next({ name: 'Unauthorized' })
+  }
+})
+
+function isUnauthorizedComponent(to) {
```

Comment:
We're really checking the destination page here. So maybe isUnauthorizedPage? We could also skip the function all together here and just do the to,name check in the beforeEach. Would be very clear what's happening and no more verbose.

Reply:
Had it as part of the if originally so moving it back there, is that what you meant?